### DISCUSSION TITLE
Feature Idea: allow create (new object) method to throw Exceptions

### DISCUSSION THREAD

**Discussion by alexanderankin** [DanielYWoo/fast-object-pool#85] [Ideas]
I wanted to use this library over ThreadLocal since virtual threads are going to render those obsolete (new thread per task = never reusing a threadlocal), and i wanted to just pass in a lambda or method reference for the create object, so i had to add this code:

```java
    private interface SimpleObjectFactory<T> extends ObjectFactory<T> {
        @Override
        default void destroy(T t) {
        }

        @Override
        default boolean validate(T t) {
            return true;
        }

        @SneakyThrows
        @Override
        default T create() {
            return doCreate();
        }

        T doCreate() throws Throwable;
    }
```

I figured it may be useful to other users of this library.

```java
    static final DisruptorObjectPool<T> TRANSFORMER = new DisruptorObjectPool<>(
            new PoolConfig(),
            (SimpleObjectFactory<T>) TheFactory.newInstance()::newT
    );
```

**Comment by DanielYWoo** [comment]
Hi @alexanderankin, I guess you do not care `validate` and `destroy`, you have a single `create` method and just want to create a Factory class with a single `create` abstract method, so you can use it as a Lambda? Can you put a unit test to show the whole thing?

**Comment by DanielYWoo** [comment]
BTW, regarding virtual threads vs ThreadLocal, I don't think it's a valid scenario to use object pools. Object pools are used for expensive objects that you don't want to create many times and destroy many times. E.g., you cannot create a JDBC connection per thread, or in a ThreadLocal, or per virtual thread (this would be the worst case and a disaster to you), none of the 3 patterns works for you because you will create too many expensive objects. Object pools are often used like you have 100 worker threads sharing 10 expensive objects.
